### DISCUSSION TITLE
find correct values for TEGRA_ variables

### DISCUSSION THREAD

**Discussion by maggu2810** [OE4T/tegra-demo-distro#242] [Q&A] [answered]
Hi, I run into a similar problem then https://github.com/OE4T/tegra-demo-distro/discussions/208

My question is, how can I get the correct values for e.g.

* TEGRA_BOARDID
* TEGRA_FAB
* TEGRA_BOARDSKU
* TEGRA_BOARDREV
* TEGRA_CHIPREV

Using `tegra-eeprom-tool` shows me

```
partnumber[nvidia]: 699-13668-0001-301 E.0
```

So, I assume this fits to:

* boardid: 3668
* fab: 301
* boardsku: 0001
* boardrev: E.0

Is this the correct way to get these values or are there better ones?
I assume all that values should be set in the custom machine configuartion file. Correct?

What about TEGRA_CHIPREV? How to get this value

Are there some more hardware specific variables that should be set for a machine?

**Comment by madisongh** [comment, marked as answer]
You generally should not need to set any of those variables unless you're adding support for a completely new module from NVIDIA that we don't already have support for.  The values there are build-time defaults that are derived by analyzing the `flash.sh` script and the `.conf` files provided in the L4T (aka Jetson Linux) BSP package.  When you flash the device, the correct binaries for the specific attributes of your hardware should get programmed, based on the actual attributes discovered during the flashing process.

For OTA bootloader updates (which is what #208 was about), the BUP payload needs to include binaries for every significant attribute variation.  We generally don't know in advance which of the possible variations are significant (it varies from module to module, and there is little to no documentation), or which of the attributes might change when NVIDIA puts out a new module.  We generally go by the contents of the `l4t_generate_soc_bup.sh` script in the BSP to determine the TEGRA_BUPGEN_SPECS settings we use.

If you're putting together a custom machine configuration for your project/product, and you know you are only ever going to use the Xavier NX 16GB module, you could change the default boardid/fab/etc. to match the model of module you have in hand, and also restrict the TEGRA_BUPGEN_SPECS list you put into your custom machine configuration to the ones for that particular model.  Just bear in mind that future revisions of that particular module model might change one of the attributes (possibly without notice), which could cause something to break later.

As for TEGRA_CHIPREV, IIRC all of the production t194 SoCs are revision 2; the actual is extracted from the ECID of the SoC (see the `tegra194-flash-helper` script, or the stock `flash.sh` script in NVIDIA's BSP).
